### ISSUE TITLE
simple_tagger: InvalidTagSequence error on Conll 2003 Dataset

### ISSUE THREAD

**Issue by manelAffi** [allenai/allennlp#3665]
Hi,
I'm trying to train the simple_tagger model for named entity recognition task on the CoNLL 2003 dataset using BERT an ELMO word embeddings. I always get the following error message:

> File "/usr/local/bin/allennlp", line 8, in <module>
    sys.exit(run())
  File "/usr/local/lib/python3.6/dist-packages/allennlp/run.py", line 18, in run
    main(prog="allennlp")
  File "/usr/local/lib/python3.6/dist-packages/allennlp/commands/__init__.py", line 102, in main
    args.func(args)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/commands/train.py", line 124, in train_model_from_args
    args.cache_prefix)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/commands/train.py", line 168, in train_model_from_file
    cache_directory, cache_prefix)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/commands/train.py", line 252, in train_model
    metrics = trainer.train()
  File "/usr/local/lib/python3.6/dist-packages/allennlp/training/trainer.py", line 478, in train
    train_metrics = self._train_epoch(epoch)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/training/trainer.py", line 320, in _train_epoch
    loss = self.batch_loss(batch_group, for_training=True)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/training/trainer.py", line 261, in batch_loss
    output_dict = self.model(**batch)
  File "/usr/local/lib/python3.6/dist-packages/torch/nn/modules/module.py", line 541, in __call__
    result = self.forward(*input, **kwargs)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/models/simple_tagger.py", line 152, in forward
    self._f1_metric(logits, tags, mask.float())
  File "/usr/local/lib/python3.6/dist-packages/allennlp/training/metrics/span_based_f1_measure.py", line 170, in __call__
    predicted_spans = tags_to_spans_function(predicted_string_labels, self._ignore_classes)
  File "/usr/local/lib/python3.6/dist-packages/allennlp/data/dataset_readers/dataset_utils/span_utils.py", line 254, in bioul_tags_to_spans
    raise InvalidTagSequence(tag_sequence)
allennlp.data.dataset_readers.dataset_utils.span_utils.InvalidTagSequence: B-ORG B-ORG B-ORG B-ORG B-ORG B-PER B-PER B-PER B-PER U-ORG B-PER B-PER B-PER B-PER B-PER U-ORG B-ORG B-PER B-PER B-PER B-PER B-PER B-PER B-PER B-PER B-PER B-PER B-PER B-ORG B-ORG B-ORG B-PER B-ORG B-ORG B-ORG

 **here is the config file code:**
`{
  "dataset_reader": {
    "type": "conll2003",
    "tag_label": "ner",
    "coding_scheme": "BIOUL",
    "token_indexers": {
      "tokens": {
        "type": "single_id",
        "lowercase_tokens": true
      },
      "token_characters": {
        "type": "characters",
        "min_padding_length": 3
      },
      "elmo": {
        "type": "elmo_characters"
     }
    }
  },
  "train_data_path": "XXXXXX",
    "validation_data_path": "XXXXXX",
    "test_data_path": "XXXXXXX",
  
  "model": {
    "type": "simple_tagger",
     "label_encoding": "BIOUL",
    
   "text_field_embedder": {
           
            "allow_unmatched_keys": true,
      
       "tokens": {
                    "type": "bert-pretrained",
                    "pretrained_model": "bert-base-cased"
                },
        "elmo":{
            "type": "elmo_token_embedder",
        "options_file": "https://allennlp.s3.amazonaws.com/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_options.json",
        "weight_file": "https://allennlp.s3.amazonaws.com/models/elmo/2x4096_512_2048cnn_2xhighway/elmo_2x4096_512_2048cnn_2xhighway_weights.hdf5",
            "do_layer_norm": false,
            "dropout": 0.0
        }
           
        
      
    },
    "encoder": {
      "type": "lstm",
      "input_size": 1792,
      "hidden_size": 100,
      "num_layers": 2,
      "dropout": 0.5,
      "bidirectional": true
    },
    "regularizer": [
      [
        "scalar_parameters",
        {
          "type": "l2",
          "alpha": 0.1
        }
      ]
    ]
  },
  "iterator": {
    "type": "basic",
    "batch_size": 32
  },
  "trainer": {
    "optimizer": {
        "type": "adam",
        "lr": 0.001
    },
    "validation_metric": "+f1-measure-overall",
    "num_serialized_models_to_keep": 3,
    "num_epochs": 3,
    "grad_norm": 5.0,
    "patience": 25,
    "cuda_device": -1
  }
}
`
**System 

OS: Google Colab
Python version: 3.6
AllenNLP version: 0.9.0
Torch version: 1.3.1

is there any ideas how to solve it?

**Comment by DeNeutoy** [comment]
Hi @manelAffi ,

This is because we don't yet have span evaluation code for ill formed sequences in the BIOUL format. If you use the `crf_tagger`, this will not happen, because the sequences are constrained to be in the correct format. We should support this, certainly - see #3169

**Comment by manelAffi** [comment]
Hi @DeNeutoy,
Thanks a lot for your response.
another question, have you any idea or tutorial how to make attention mechanism between word embedding with allennlp?
Thank you,

**Comment by DeNeutoy** [comment]
Hi, you might like to look at our docs, particularly the attention mechanisms: https://allenai.github.io/allennlp-docs/api/modules/attention/attention/

or you could also ask on our discussion forum, discourse.allennlp.org
